### PR TITLE
chore: fix deepsource issues found in release v0.6.0

### LINKED ISSUE #1729: Fix DeepSource issues detected for Release V0.6.0
## Task summary

During merge of development into main for release v0.6.0, DeepSource analysis detected the following issues:
- [x] Unused code in dot/network/host.go starting at line 228
- [x] Return of `nil` value detected in lib/scale/encode.go at line 128
- [x] Shadow of builtin detected in dot/rpc/service.go at line 94
- [ ] Function call made to an unsafe package in dot/network/notifications.go at line 32
- [x] Audit the random number generation source (rand) in:
     - lib/trie/test_utils.go line 42
     - dot/state/test_helpers.go line 64
     - dot/network/utils.go line 71
     - dot/network/connmgr.go line 166
- [ ] Empty body in an if or else branch in dot/rpc/modules/grandpa.go line 80

## Specification reference

- see https://deepsource.io/gh/ChainSafe/gossamer/run/52c1d515-ffe8-4cf8-a9f0-324143e289cf/go/ for details.

## Other information and links

-

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Fixes issues found by DeepSource during merge of release v0.6.0.
- Did not address the following issues:
    - Function call made to unsafe.SizeOf, I'm not sure how to replace that functionality
    - Replace math/rand with crypto/rand, did not change in dot/network/utils.go line 71 because that function requires the deterministic randomness of math/rand.
    - Empty body, did not remove this because there are comments referencing an issue in the body.

## Tests

```

```

## Issues

- closes #1729 

## Primary Reviewer 

- @arijitAD

### CODE REVIEW

File: dot/network/connmgr.go [hunk 1]
```diff
@@ -162,13 +163,15 @@ func (cm *ConnManager) Connected(n network.Network, c network.Conn) {
 			return
 		}
 
-		// TODO: change to crypto/rand
-		i := rand.Intn(len(unprotPeers)) //nolint
+		i, err := rand.Int(rand.Reader, big.NewInt(int64(len(unprotPeers))))
+		if err != nil {
+			logger.Error("error generating random number", "error", err)
```
Comment (timwu20) [inline dot/network/connmgr.go:168]: should we continue if this happens?

File: dot/rpc/service.go [hunk 2]
```diff
@@ -91,8 +91,8 @@ func (s *Service) BuildMethodNames(rcvr interface{}, name string) {
 
 // isExported returns true of a string is an exported (upper case) name.
 func isExported(name string) bool {
-	rune, _ := utf8.DecodeRuneInString(name)
-	return unicode.IsUpper(rune)
+	r, _ := utf8.DecodeRuneInString(name)
```
Comment (timwu20) [inline dot/rpc/service.go:94]: why do we not check this error?

File: lib/trie/test_utils.go [hunk 3]
```diff
@@ -39,14 +40,17 @@ func GenerateRandomTests(t testing.TB, size int) []Test {
...
-	r := *rand.New(rand.NewSource(rand.Int63())) //nolint
 	test := Test{}
 
 	for {
 		n := 2 // arbitrary positive number
-		size := r.Intn(510) + n
-		buf := make([]byte, size)
-		_, err := r.Read(buf)
+		size, err := rand.Int(rand.Reader, big.NewInt(510))
+		if err != nil {
```
Comment (arijitAD) [inline lib/trie/test_utils.go:48]: `require.NoError(t, err)`
Also, replace in other places.

File: dot/network/connmgr.go (hunk 1 above)
Comment (edwardmack) [reply dot/network/connmgr.go:168]: probably not, I added a return here.

File: dot/network/connmgr.go [hunk 4]
```diff
@@ -162,13 +163,15 @@ func (cm *ConnManager) Connected(n network.Network, c network.Conn) {
...
+		}
 
-		logger.Trace("Over max peer count, disconnecting from random unprotected peer", "peer", unprotPeers[i])
-		err := n.ClosePeer(unprotPeers[i])
+		logger.Trace("Over max peer count, disconnecting from random unprotected peer", "peer", unprotPeers[i.Int64()])
```
[the remark this answers is not in the document]
Comment (edwardmack) [reply dot/network/connmgr.go:171]: add separate variable.

File: dot/rpc/service.go (hunk 2 above)
Comment (edwardmack) [reply dot/rpc/service.go:94]: utf8.DecodeRuneInString returns rune and size, the second ignored var is for size not for error.

File: lib/trie/test_utils.go (hunk 3 above)
Comment (edwardmack) [reply lib/trie/test_utils.go:48]: updated.

**Review by arijitAD — APPROVED** [review]

**Review by EclesioMeloJunior — APPROVED** [review]

**Review by timwu20 — APPROVED** [review]

**Review by noot — APPROVED** [review]

**Pull request merged**